gitlab/resource_gitlab_project.go

```diff
@@ -191,6 +195,83 @@ var resourceGitLabProjectSchema = map[string]*schema.Schema{
 		Optional: true,
 		Default:  true,
 	},
+	"push_rules": {
+		Type:     schema.TypeList,
+		MaxItems: 1,
+		Optional: true,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"author_email_regex": {
+					Type:         schema.TypeString,
+					Optional:     true,
+					ValidateFunc: validation.StringIsValidRegExp,
+				},
+				"branch_name_regex": {
+					Type:         schema.TypeString,
+					Optional:     true,
+					ValidateFunc: validation.StringIsValidRegExp,
+				},
+				"commit_message_regex": {
+					Type:         schema.TypeString,
+					Optional:     true,
+					ValidateFunc: validation.StringIsValidRegExp,
+				},
+				"commit_message_negative_regex": {
+					Type:         schema.TypeString,
+					Optional:     true,
+					ValidateFunc: validation.StringIsValidRegExp,
+				},
+				"file_name_regex": {
+					Type:         schema.TypeString,
+					Optional:     true,
+					ValidateFunc: validation.StringIsValidRegExp,
```

Comment:
That validation method: are the regex of Go exactly overlapping with Ruby regexes??

Reply:
I thought only Javascript was different? Do you want me to remove validation here?

Reply:
no, golang is opiniated.

e.g. they do not support backreference, whereas ruby does.

Reply:
Good to know. I've removed the validation.